rfcs/0007_reorganize_repositories.md

```diff
@@ -0,0 +1,74 @@
+- Start Date: 2025-01-28
+- RFC PR: https://github.com/thousandbrainsproject/tbp.monty/pull/159/
+
+# Summary
+
+The proposal is to establish a better way of organizing Thousand Brains Project repositories.
+
+Currently there are two repositories:
+
+1. `tbp.monty` - This is the main repository for the Monty framework.
+2. `monty_lab` - This is what `nupic.monty/projects` used to be, and is currently a catch-all for code that is not part of the main Monty framework.
+
+The high-level proposal is to break-up `monty_lab` where appropriate, and have a lower threshold for creating new, independent repositories.
+
+More specifically, the proposed structure and description of each repository would be:
+
+`tbp.monty`
+- Core code-base for the Monty framework.
+- All code has undergone thorough review, and Circle-CI runs unit-tests and style checks on any new code.
+- Should not become bloated with experimental code that is unlikely to be re-used in future work.
+
+`monty_lab`
+- A repository to house discontinued or paused research projects like  `high_dim_coincidence_detection` and `grid_cells`.
+- Local forks of `monty_lab` can be used by contributors to track in-progress work. However, this does not mean that such code, once mature enough to be shared, should necessarily be merged into the main `monty_lab` repository.
+- In general, if the code is likely to be re-used in the future, or forms part of a paper, it should *not* go in `monty_lab` (more on this below).
+- Circle-CI does not include style or unit-test checks on new code. 
+- When merging a discontinued project, PR review does not need to be as thorough as for code pushed to the main `tbp.monty` repository.
+- Not intended as a code-base that community members can contribute to. As such, issues like bugs are less likely to be noticed and fixed. 
+
+We could then have two additional types of repositories, with no predefined number of these. These repository *types* would be:
```

Comment:
I would suggest to just have one type of repository (structure could be defined by a template repository). There will inevitable be more usecases for repositories that these two (like the plotting repo that Will made, an exploratory analysis repo, a repo for experiments on a specific topic that was not published as a paper,...) The name and readme of the repo should be sufficient to inform people of the content.

Reply:
Ok yeah, this is maybe the one point I'm on the fence about. Couldn't most of those fit well within `monty_lab`? Personally I feel like having some prefix in the name to clarify it's purpose is useful, but maybe that's just me.